controllers/booksController.js

```diff
@@ -18,15 +18,29 @@ export const handleNewBook = async (req, res) => {
     const result = await Book.create({
       'title': title,
       'author': author,
-      'readOn': readOn
+      'readOn': readOn,
+      'currentlyReading': currentlyReading
     });
     res.status(201).json({ 'message': `New book ${title} added!` });
   } catch (err) {
     res.status(500).json({ 'message': err.message });
   }
 };
 
-export const handleGetBooks = async (req, res) => {
+export const getRandomBook = async (req, res) => {
+  let selection;
+  try {
+    const books = await toBeRead();
+    if (books) {
+      selection = books[Math.floor(Math.random() * books.length)]
```

Comment:
i believe you want `(books.lenghth - 1)`. If books.length is 10, this could return 10 as an index which isn't valid.